From 50544d21e8464f04d17366b197d720f8377cdaa5 Mon Sep 17 00:00:00 2001
From: Jeremy Gebben <jgebben@codeaurora.org>
Date: Tue, 27 Dec 2011 03:48:00 +0100
Subject: [PATCH] msm: kgsl: put gpu_id into struct kgsl_devinfo

The field gmem_hostbaseaddr has never been used by userspace
and has always been set to 0. It is being replaced with the
gpu_id field which will contain the revision of the GPU,
for example 200 for adreno200.

Change-Id: I48347da5d430a3aa1db588b4fe54febe7eb44476
Signed-off-by: Jeremy Gebben <jgebben@codeaurora.org>
---
 .../video/msm/gpu/kgsl_adreno205_hc/kgsl_yamato.c  |    4 ++--
 include/linux/msm_kgsl.h                           |   10 ++++++----
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl_yamato.c b/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl_yamato.c
index 580acd4..ba0cef5 100644
--- a/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl_yamato.c
+++ b/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl_yamato.c
@@ -826,8 +826,8 @@ static int kgsl_yamato_getproperty(struct kgsl_device *device,
 			devinfo.device_id = device->id+1;
 			devinfo.chip_id = device->chip_id;
 			devinfo.mmu_enabled = kgsl_mmu_isenabled(&device->mmu);
-			devinfo.gmem_hostbaseaddr = (unsigned int)
-					yamato_device->gmemspace.mmio_virt_base;
+//			devinfo.gpu_id = adreno_get_rev(adreno_dev);
+			devinfo.gpu_id = 205;
 			devinfo.gmem_gpubaseaddr = yamato_device->gmemspace.
 					gpu_base;
 			devinfo.gmem_sizebytes = yamato_device->gmemspace.
diff --git a/include/linux/msm_kgsl.h b/include/linux/msm_kgsl.h
index c0ac369..57c971a 100644
--- a/include/linux/msm_kgsl.h
+++ b/include/linux/msm_kgsl.h
@@ -30,7 +30,7 @@
 #define _MSM_KGSL_H
 
 #define KGSL_VERSION_MAJOR        3
-#define KGSL_VERSION_MINOR        3
+#define KGSL_VERSION_MINOR        4
 
 /*context flags */
 #define KGSL_CONTEXT_SAVE_GMEM	1
@@ -89,9 +89,11 @@ struct kgsl_devinfo {
 	unsigned int chip_id;
 	unsigned int mmu_enabled;
 	unsigned int gmem_gpubaseaddr;
-	/* if gmem_hostbaseaddr is NULL, we would know its not mapped into
-	 * mmio space */
-	unsigned int gmem_hostbaseaddr;
+	/*
+	* This field contains the adreno revision
+	* number 200, 205, 220, etc...
+	*/
+	unsigned int gpu_id;
 	unsigned int gmem_sizebytes;
 };
 
-- 
1.7.5.4

